boards/risc-v/rv32m1/rv32m1-vega/src/rv32m1_buttons.c

```diff
@@ -0,0 +1,151 @@
+/****************************************************************************
+ * boards/risc-v/rv32m1/rv32m1-vega/src/rv32m1_buttons.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/board.h>
+#include <nuttx/arch.h>
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <syslog.h>
+#include <errno.h>
+
+#include "rv32m1_gpio.h"
+#include "rv32m1.h"
+#include "rv32m1-vega.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const uint32_t g_buttons[] =
+{
+  GPIO_INPUT|GPIO_INT_EDGE|GPIO_FLOAT |GPIO_PORTA|GPIO_PIN0,  /* SW2 */
+  GPIO_INPUT|GPIO_INT_EDGE|GPIO_PULLUP|GPIO_PORTE|GPIO_PIN8,  /* SW3 */
+  GPIO_INPUT|GPIO_INT_EDGE|GPIO_PULLUP|GPIO_PORTE|GPIO_PIN9,  /* SW4 */
+  GPIO_INPUT|GPIO_INT_EDGE|GPIO_PULLUP|GPIO_PORTE|GPIO_PIN12, /* SW5 */
```

Comment:
Please define the buttons names and just put them here, please don't define it directly here inside the array.